instrumentation/opentelemetry_bandit/lib/opentelemetry_bandit.ex

```diff
@@ -1,178 +1,515 @@
+                      doc: "Custom client header sort fn. See `otel_http` for more info"
+                    ],
+                    public_endpoint: [
+                      type: :boolean,
+                      default: false,
+                      doc: "Endpoint is public. Propagated traces will be added as a link."
+                    ],
+                    public_endpoint_fn: [
+                      type: :mfa,
+                      default: {__MODULE__, :default_public_endpoint_fn, []},
+                      doc: "Default function returns `false`. See docs for more info"
+                    ],
+                    request_headers: [
+                      type: {:list, :string},
+                      default: [],
+                      doc: "List of request headers to add as attributes. (lowercase)"
+                    ],
+                    response_headers: [
+                      type: {:list, :string},
+                      default: [],
+                      doc: "List of response headers to add as attributes. (lowercase)"
+                    ],
+                    scheme_headers: [
+                      type: {:list, :string},
+                      default: ["forwarded", "x-forwarded-proto"],
+                      doc: "Headers to use for extracting original client request scheme"
+                    ],
+                    scheme_headers_sort_fn: [
+                      type: {:fun, 2},
+                      doc: "Custom scheme header sort fn. See `otel_http` for more info"
+                    ],
+                    server_address_headers: [
+                      type: {:list, :string},
+                      default: ["forwarded", "x-forwarded-host", "host"],
+                      doc: "Headers to use for extracting original server address info"
+                    ],
+                    server_headers_sort_fn: [
+                      type: {:fun, 2},
+                      doc: "Custom server header sort fn. See `otel_http` for more info"
+                    ]
+                  )
+
+  @type options() :: [unquote(NimbleOptions.option_typespec(@options_schema))]
+
+  @moduledoc """
+  OpenTelemetry instrumentation for Bandit.
+
+  ## Semantic Conventions
+
+  All required and recommended Server HTTP Span semantic conventions are implemented.
+  Supported opt-in attributes can be configured using the `opt_in_attrs` option.
+
+  ## Options
+
+  ### Opt-in Semantic Convention Attributes
+
+  Otel SemConv requires users to explicitly opt in for any attribute with a
+  requirement level of `opt-in`. To ensure compatability, always use the
+  SemConv attribute.
+
+  Example:
+  ```
+  opt_ins = [{SemConv.HTTPAttributes.http_request_body_size(), true}]`
+  OpentelemetryBandit.setup(opt_in_attrs: opt_ins)
+  ```
+
+  Request and response header attributes are opt-in and can be set with the
+  `request_headers` and `response_headers` options. Values should be lower-case.
+
```

Comment:
I feel like this section needs its own heading since these options are _also_ kind of opt-in, but they're not configured in the `opt_in_attrs` option being described above (right?)

```suggestion
  ### Request and Response Headers as Span Attributes

  Request and response header attributes are opt-in and can be configured with the
  `request_headers` and `response_headers` options. Header namess should be lower-case.

  Example:
  ` ` ` <- GitHub suggestions won't let me include triple quotes
  OpentelemetryBandit.setup(request_headers: ["x-customer-id"])
  ` ` `

```